readthedocs/projects/tasks.py

```diff
@@ -587,7 +587,7 @@ def run_build(self, docker, record):
                 log.warning('No build ID, not syncing files')
 
         if self.build_env.failed:
-            self.send_notifications()
+            self.send_notifications(self.version.pk, self.build['id'])
```

Comment:
Some lines above we are using `self.build.get('id')` and then `if build_id`. So, I suppose that we can't always access the `id` like here and we should probably use `build_id` directly which could be `None`.